components/pipedrive/sources/new-deal/new-deal.mjs

```diff
@@ -0,0 +1,45 @@
+import common from "../common/base.mjs";
+import constants from "../../common/constants.mjs";
+
+export default {
+  ...common,
+  key: "pipedrive-new-deal",
+  name: "New Deal",
+  description: "Triggers when a new deal is created.",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  methods: {
+    ...common.methods,
+    getResourceFn() {
+      return this.pipedriveApp.getDeals;
+    },
+    getResourceFnArgs() {
+      return {
+        sort: "id DESC",
+      };
+    },
+    getResourceProperty() {
+      return "id";
+    },
+    getEventObject() {
+      return constants.EVENT_OBJECT.DEAL;
+    },
+    getEventAction() {
+      return constants.EVENT_ACTION.ADDED;
+    },
+    generateMeta(resource) {
+      return {
+        id: resource.id,
+        summary: `${this.getEventObject()} ${resource.id} was ${this.getEventAction()}`,
+        ts: Date.parse(resource.add_time),
+      };
+    },
+    done({
+      resource, lastResourceProperty,
+    }) {
+      const property = this.getResourceProperty();
+      return lastResourceProperty === String(resource[property]);
+    },
```

Comment:
If we move `done` and `generateMeta` to `base.mjs`, this would change to:
```suggestion
    getTimestamp(resource) {
      return Date.parse(resource.add_time);
    },
```